File: frontend/types/produto.ts
export interface ProdutoRequest {
  nome: string;
  qtdEstoque: number;
  preco: number;
  custo?: number;
  dataAquisicao?: string; // ISO date string
  fornecedor?: string;
}

export interface ProdutoResponse {
  id: number;
  nome: string;
  qtdEstoque: number;
  preco: number;
  custo?: number;
  dataAquisicao?: string; // ISO date string
  fornecedor?: string;
}

File: backend/src/main/java/br/com/julia/rgsurfschool/domain/model/Produto.java
package br.com.julia.rgsurfschool.domain.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity(name = "tb_produto")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column(nullable = false)
    String nome;

    @Column(name = "qtd_estoque", nullable = false)
    Integer qtdEstoque;

    @Column(nullable = false, precision = 10, scale = 2)
    BigDecimal preco;

    @Column(precision = 10, scale = 2)
    BigDecimal custo;

    @Column(name = "data_aquisicao")
    LocalDate dataAquisicao;

    @Column
    String fornecedor;

}

